service/nodes.go

```diff
@@ -0,0 +1,120 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// nolint:unused
+package service // import "go.opentelemetry.io/collector/service"
+
+import (
+	"hash/fnv"
+	"strings"
+
+	"gonum.org/v1/gonum/graph"
+
+	"go.opentelemetry.io/collector/component"
+)
+
+type nodeID int64
+
+func (n nodeID) ID() int64 {
+	return int64(n)
+}
+
+func newNodeID(parts ...string) nodeID {
```

Comment:
This looks a bit "lost" in this PR -- where is it being used?

Reply:
It's used by each of the `new<ComponentType>Node` functions, to generate their unique ids.